android/src/main/java/com/liferay/mobile/android/auth/refresh/OAuth2AuthenticationRefreshHandler.java

```diff
@@ -0,0 +1,75 @@
+package com.liferay.mobile.android.auth.refresh;
+
+import android.provider.Settings;
+import com.liferay.mobile.android.auth.CookieSignIn;
+import com.liferay.mobile.android.auth.OAuth2SignIn;
+import com.liferay.mobile.android.auth.SessionCallback;
+import com.liferay.mobile.android.auth.basic.CookieAuthentication;
+import com.liferay.mobile.android.auth.oauth2.OAuth2Authentication;
+import com.liferay.mobile.android.service.Session;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class OAuth2AuthenticationRefreshHandler implements AuthenticationRefreshHandler {
+
+	@Override
+	public Session refreshAuthentication(final Session session, final SessionCallback callback) throws Exception {
```

Comment:
Can we not throw an exception here? I'm afraid missing an exception somewhere and not unlocking the lock.

Reply:
I agree, I would wrap the second call in a try just in case